src/constraint_solver/constraint_solver.h

```diff
@@ -2916,6 +2917,10 @@ class Solver {
   // to the solver before the search.
   int constraints() const { return constraints_list_.size(); }
 
+  //// TODO: TBD(MWP): may expose these collections eventually, but not at the moment. at minimum be able to reference constraints by name
+  //const std::vector<Constraint*>* get_constraint_list() { return &constraints_list_; }
```

Comment:
I definitely do not want to expose this., Especially the additional constraint list.